assignments/prototypes.js

```diff
@@ -41,70 +109,203 @@
+  name: 'Bruce',
+  team: 'Mage Guild',
+  weapons: [
+    'Staff of Shamalama',
+  ],
+  language: 'Common Tongue',
+});
+
+const swordsman = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 2,
+    height: 2,
+  },
+  healthPoints: 15,
+  name: 'Sir Mustachio',
+  team: 'The Round Table',
+  weapons: [
+    'Giant Sword',
+    'Shield',
+  ],
+  language: 'Common Tongue',
+});
+
+const archer = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 1,
+    width: 2,
+    height: 4,
+  },
+  healthPoints: 10,
+  name: 'Lilith',
+  team: 'Forest Kingdom',
+  weapons: [
+    'Bow',
+    'Dagger',
+  ],
+  language: 'Elvish',
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+/**
+ * Hero and Villian Constructors
+ */
+function Hero(feat) {
+  Humanoid.call(this, feat);
+  this.powerStat = feat.powerStat;
+}
+
+Hero.prototype = Object.create(Humanoid.prototype);
+
+function Villain(feat) {
+  Humanoid.call(this, feat);
+  this.powerStat = feat.powerStat;
+}
+
+Villain.prototype = Object.create(Humanoid.prototype);
+Humanoid.prototype.speak = function() {
+  return `Hi I am ${this.name} the leader of ${this.team}.`;
+}
+Hero.prototype.attack = function(person) {
+  if(this.powerStat.durability + this.powerStat.combat +this.powerStat.power + this.powerStat.strength > 30) {
+    person.healthPoints -= 30;
+    return `${person.name} health point is now ${person.healthPoints}`
+  } else {
+    person.healthPoints -= 10;
+    return `${person.name} health point is now ${person.healthPoints}`
+  }
+}
+
+Villain.prototype.attack = function(person) {
+  if(this.powerStat.durability + this.powerStat.combat +this.powerStat.power + this.powerStat.strength > 30) {
+    person.healthPoints -= 30;
+    return `${person.name} health point is now ${person.healthPoints}`
+  } else {
+    person.healthPoints -= 10;
+    return `${person.name} health point is now ${person.healthPoints}`
+  }
+}
```

Comment:
As these two prototypes are exactly the same it's possible to create a separate function and assign it to the attack prototype of hero and villain, something like this:

Hero.prototype = { ...Humanoid.prototype, attackFunction };